ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v1/container/clientlibs/editor/js/editDialog.js

```diff
@@ -42,6 +42,259 @@
+        $.ajax({
+            type: "POST",
+            url: Granite.HTTP.externalize("/libs/fd/fm/gui/content/forms/guide/validate.html?xsdFilePath=" + encodeURIComponent(path) + "&assetType=guide")
+        }).done(function(html){
+            var autocomplete =  getXsdRootElementDropDown(html, preselectedRootElement);
+            var label = '<label class="coral-Form-fieldlabel">' + Granite.I18n.get("XML Schema Root Element *") + '</label>';
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).html(label);
+            $(SCHEMA_ROOT_ELEMENT_SELECTOR).append($(autocomplete));
+            $(SCHEMA_DROPDOWN_SELECTOR).val(selectedSchema);
+        }).fail(function(e){
+            console.log(e);
+        });
+    };
+
+    function getXsdRootElementDropDown(json, preselectedRootElement){
+        var placeHolder = Granite.I18n.get("Select or type a Root Element");
+        removeXsdRootElementMarkup();
+        var autocomplete = new Coral.Autocomplete().set({
+            placeholder : placeHolder,
+            required : true
+        });
+        autocomplete.forceSelection = true;
+        $(autocomplete).addClass('rootSelect');
+        for(var i = 0; i < json.length; i++){
+            var key = Object.keys(json[i]);
+            var value = json[i][key];
+            var selected = preselectedRootElement == value;
+            autocomplete.items.add({
+                value : _g.shared.XSS.getXSSValue(value),
+                content:
+                    {
+                        innerHTML : _g.shared.XSS.getXSSValue(key)
+                    },
+                selected: selected
+            });
+        }
+        return autocomplete;
+    };
+
+    function fillXsdDetails(xsdRef) {
+        var xsdRootElement = $(XSD_ROOT_ELEMENT_SELECTOR).val();
+        createRootElementSelectionMarkup(xsdRef, xsdRootElement);
+    };
+
+    /**
+     * Clears the ./xsdRootElement attr and deletes the rootElement dropdown markup
+     */
+    function removeXsdRootElementMarkup(){
+        $(XSD_ROOT_ELEMENT_SELECTOR).val('');
+        const elements = document.getElementsByClassName('rootSelect');
+        while(elements.length > 0){
+            elements[0].parentNode.removeChild(elements[0]);
+        }
+    };
+
+    function hideContainersExcept(selectedSchemaType) {
+        if (selectedSchemaType == SCHEMA || selectedSchemaType == JSON_SCHEMA || selectedSchemaType == XML_SCHEMA) {
+            $(FDM_CONTAINER).hide();
+            $(SCHEMA_CONTAINER).show();
+        } else if (selectedSchemaType == FORM_DATA_MODEL) {
+            $(SCHEMA_CONTAINER).hide();
+            $(FDM_CONTAINER).show();
+        } else if (selectedSchemaType == 'none') {
+            $(FDM_CONTAINER).hide();
+            $(SCHEMA_CONTAINER).hide();
+        }
+    };
+
+    function isTemplate() {
+        var currentUrl = window.location.href;
+        return !currentUrl.includes("editor.html/content/forms/af/");
```

Comment:
we need a better contract. relying on urls is hacky.

Reply:
isTemplate() check was added in AF v1 to align with V1 where if one type of schema is selected, then authors are not allowed to a different type, For instance if xsd type was already selected, then authors couldn't update to json. But it was later discussed that this check was redundant. And if we are not using xml schema anyway then I think we really don't need this check . cc :- @vdua

Reply:
ok, removed it